src/audio/kpb.c

```diff
@@ -225,6 +227,16 @@ static int kpb_prepare(struct comp_dev *dev)
 	/* initialie draining task */
 	schedule_task_init(&cd->draining_task, kpb_draining_task, cd);
 
+	/* search for the channel selector sink */
+	list_for_item(blist, &dev->bsink_list) {
+		sink = container_of(blist, struct comp_buffer, source_list);
+		if (sink->sink->comp.type == SOF_COMP_SELECTOR) {
```

Comment:
Agree, best to make it obvious this is temporary in the comments as it implies a fixed topology. Ideally we should have a token from the topology to describe the correct sink.

Reply:
@tlauda The code will be removed as part of optimizing resources used by the detect pipeline I hope, however h/c for the selector could be optimal and right choice till then, since the detect pipeline is expected to process minimal amount of data required, no matter what additional components are there, therefore selector usually would go first.

Reply:
Updated.